src/components/button.tsx

```
@@ -1,11 +1,45 @@
-interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+import type { MouseEvent } from 'react';
+
+type ButtonClickHandler =
+  | ((...args: any[]) => void | Promise<void | any>)
+  | ((event: MouseEvent<HTMLButtonElement>) => void | Promise<void>);
+
+interface ButtonProps extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'onClick'> {
   isLoading?: boolean;
   isLoadingText?: string;
+  onClick?: ButtonClickHandler;
 }
 
-export function Button({ children, disabled, isLoading, isLoadingText, ...props }: ButtonProps) {
+export function Button({
+  children,
+  disabled,
+  isLoading,
+  isLoadingText,
+  onClick,
+  ...props
+}: ButtonProps) {
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    if (onClick && !disabled && !isLoading) {      
+      try {
```

I don't understand what's going on here. First, every instance of `Button` seems to already use a try block for the `onClick` event. Second, why not just try catch the onClick call itself then if the idea is to silence the error?

---

When using Reown AppKit, it offers a connect function. When I passed this function directly, I encountered type errors in TypeScript, so I needed to customize the button to also accept this type of connection function.

---

I tried reverting this, and except within `connect-wallet.ts` where the button arguments can be changed to `onClick={() => open}` everything works fine